### PR TITLE
Fix llama_factory_llama3 notebook: pin LLaMA-Factory install, align model references

### PR DESCRIPTION
**Pull request by sabira-mcw**
## Motivation

The `docs/notebooks/fine_tune/llama_factory_llama3.ipynb` tutorial cannot currently be run end-to-end. Three independent issues each stop it at a different point:

1. The install cell aborts, because upstream LLaMA-Factory now requires Python >= 3.11 while the tutorial container ships Python 3.10.
2. The training cell exits during argument parsing, because LLaMA-Factory requires DeepSpeed recipes to be launched through `torchrun`.
3. The notebook's prose and Hugging Face access instructions reference Llama-3.1 8B, while the recipe it runs trains Llama-3 8B. These are separately gated repositories, so a reader who follows the access instructions can still be denied at training time.

The goal is a notebook that runs start to finish on a current ROCm PyTorch container without manual intervention.

## Technical Details

**Install (issue 1).** The cell cloned `main` unpinned. Upstream dropped Python 3.10 in [hiyouga/LLaMA-Factory#9677](https://github.com/hiyouga/LLaMA-Factory/pull/9677) (`requires-python = ">=3.11.0"`), so `pip install -e .` now fails on the container's Python 3.10.

Pinning to v0.9.3 - the last tag declaring `python_requires=">=3.9.0"` - installs, but then fails at training: that tag's `examples/train_full/llama3_full_sft.yaml` references `examples/deepspeed/ds_z3_config.json`, and `examples/deepspeed/` is not present in the v0.9.3 tree.

**Training launch (issue 2).** Added `FORCE_TORCHRUN=1` to the training command, as required by `src/llamafactory/hparams/parser.py` for any recipe specifying `deepspeed`.

**Model references (issue 3).** Title, introduction, Hugging Face access links, token section, and download cell now all reference
`meta-llama/Meta-Llama-3-8B-Instruct`, matching `model_name_or_path` in `llama3_full_sft.yaml`.

**Documentation corrections.**

- Removed "fine-tune Llama-3 8B on eight GPUs" - the command shown passes no `--nproc_per_node`; LLaMA-Factory uses whatever GPUs it detects.
- Added a note on checkpoint size: the recipe saves every 500 steps including optimizer states, so readers should set `save_only_model` / `save_steps` to suit available disk.
- Moved GPU monitoring to `watch -n0 amd-smi` in a terminal, since the training cell blocks the kernel and an in-notebook `!amd-smi` cannot run while training is active.
- Expanded the install section to describe DeepSpeed and LLaMA-Factory.
- Added a Conclusion section.

## Test Plan

Executed the notebook top to bottom on a single AMD Instinct MI300X (192 GB):

- ROCm 7.2, PyTorch 2.8, Python 3.10.18
- Verified LLaMA-Factory installs from the pinned commit and reports `0.9.4.dev0`
- Verified `examples/deepspeed/ds_z3_config.json` is present in the pinned checkout
- Verified the model downloads with the Hugging Face token flow as documented
- Ran the full-parameter SFT recipe and monitored loss and VRAM

## Test Result

- Install completes on Python 3.10 with no dependency resolution errors.
- Training launches through `torchrun` and runs the recipe's 1635 steps with loss decreasing normally.
- Peak VRAM ~130 GB, within the MI300X's 192 GB on a single device.

### CODE REVIEW

File: docs/notebooks/fine_tune/llama_factory_llama3.ipynb [hunk 1]
```diff
@@ -388,33 +402,29 @@
...
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "45fb36ff",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "!amd-smi"
+    ">**watch -n0 amd-smi**\n",
+    "\n",
+    "**Note**: For ROCm 6.4 and earlier, use the `rocm-smi` command instead.\n",
```
Comment (edamamez) [inline docs/notebooks/fine_tune/llama_factory_llama3.ipynb]: Since the container is using 6.3, lets swap these
```suggestion
    ">**watch -n0 rocm-smi**\n",
    "\n",
    "**Note**: For ROCm 6.5 and later, use the `amd-smi` command instead.\n",
```
